File: src/com/suushiemaniac/cubing/bld/model/enumeration/TetrahedronPieceType.java
package com.suushiemaniac.cubing.bld.model.enumeration;

import com.suushiemaniac.cubing.alglib.lang.NotationReader;
import com.suushiemaniac.cubing.alglib.lang.PyraminxAlgorithmReader;

public enum TetrahedronPieceType implements PieceType {
    TIP, CENTER, EDGE;

    public static final NotationReader READER = new PyraminxAlgorithmReader();

    @Override
    public int getTargetsPerPiece() {
        return 0;
    }

    @Override
    public int getNumPieces() {
        return 0;
    }

    @Override
    public int getNumPiecesNoBuffer() {
        return 0;
    }

    @Override
    public int getNumAlgs() {
        return 0;
    }

    @Override
    public NotationReader getReader() {
        return TetrahedronPieceType.READER;
    }

    @Override
    public String humanName() {
        switch (this) {
            case TIP:
                return "Tip";
            case CENTER:
                return "Center";
            case EDGE:
                return "Edge";
            default:
                return "";
        }
    }

    @Override
    public String mnemonic() {
        switch (this) {
            case TIP:
                return "T";
            case CENTER:
                return "C";
            case EDGE:
                return "E";
            default:
                return "";
        }
    }
}

File: src/com/suushiemaniac/cubing/bld/model/enumeration/piece/TetrahedronPieceType.kt
package com.suushiemaniac.cubing.bld.model.enumeration.piece

import com.suushiemaniac.cubing.alglib.lang.NotationReader
import com.suushiemaniac.cubing.alglib.lang.PyraminxAlgorithmReader

enum class TetrahedronPieceType(override val targetsPerPiece: Int, override val numPieces: Int, override val numAlgs: Int, override val humanName: String, override val mnemonic: String) : PieceType {
    TIP(3, 4, 0, "Tip", "T"),
    CENTER(3, 4, 0, "Center", "C"),
    EDGE(2, 6, 80, "Edge", "E");

    override val reader: NotationReader
        get() = READER

    companion object {
        val READER: NotationReader = PyraminxAlgorithmReader()
    }
}
